src/reducers/AddSiteReducer.js

```diff
@@ -74,14 +95,32 @@ export default (state = INITIAL_STATE, action) => {
             return {...state, facilitiesOption};
 
         case ADD_SITE_FIELDS_RESET:
-            return INITIAL_STATE;
+            return {...INITIAL_STATE, addSiteModalVisible: true};
 
         case CURRENT_LOCATION_UPDATED:
             const {currentLocation: {longitude, latitude}} = payload;
-            const stringLong =  _.toString(longitude);
+            const stringLong = _.toString(longitude);
             const stringLat = _.toString(latitude);
 
-            return {...state, longitudeText: stringLong, latitudeText: stringLat};
+            return {
+                ...state,
+                longitudeText: stringLong,
+                latitudeText: stringLat,
+                readyLatitude: latitude,
+                readyLongitude: longitude
+            };
+
+        case CHECK_IF_SITE_IS_READY:
+            const siteReadyForUpload = siteIsReadyForUpload(state);
+
+            return {...state, siteReadyForUpload};
```

Comment:
This is the use of the `siteIsReadyForUpload` function I mentioned before. This reducer is triggered by an action, which in turn is triggered every time the `AddSiteScreen` mounts or receives any props. (This way we catch any case of anything loading from scratch with a complete state, or when anything changes). More on that when we get to the `AddSiteScreen`.